src/main/java/com/nowoczesnyjunior/financialapp/expensemodule/repository/ExpenseRepository.java

```diff
@@ -0,0 +1,25 @@
+package com.nowoczesnyjunior.financialapp.expensemodule.repository;
+
+import com.nowoczesnyjunior.financialapp.expensemodule.model.Expense;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Repository
+public interface ExpenseRepository extends JpaRepository<Expense, Long> {
+
+    List<Expense> findExpenseByExpenseDateBetween(LocalDateTime startDate, LocalDateTime endDate);
+
+    @Query("SELECT e FROM Expense e INNER JOIN e.category ec " +
+            "WHERE ec.categoryName = :categoryName " +
+            "AND e.expenseDate BETWEEN :startDate AND :endDate")
+    List<Expense> findExpenseByExpenseDateBetweenAndCategory(
```

Comment:
Is query required here? Wouldn't it work with Jpa query method? Like findExpenseByExpenseDateBetweenAndCategory_CategoryName

Reply:
done

Reply:
done :) I didn't realize that it can handle joining other queries - like category name is from other table, nice :D